lib/runner/mocha.js

```diff
@@ -114,18 +114,15 @@ function MochaRunner(runnerConfig) {
     mocha.addFile(file);
   });
 
-  if (input.profile.conf.parallel === 'file' && mocha.options.grep) {
-    // TODO: can this be a more generalized check for zero tests?
-    // if we're running parallel by file, make sure we really need to run()
-    mocha.loadFiles();
-    const throwawayRunner = new Mocha.Runner(mocha.suite);
-    const matchingTests = throwawayRunner.grep(mocha.options.grep).total;
-    if (!matchingTests) {
-      // don't run the suite, just exit the process
-      log('no tests. exiting');
-      mochaCompleteResolvers.resolve(testResults);
-      return mochaCompletePromise;
-    }
+  log('look for matching tests based on instance configuration');
+  mocha.loadFiles();
+  const throwawayRunner = new Mocha.Runner(mocha.suite);
+  const matchingTests = throwawayRunner.grep(mocha.options.grep).total;
```

Comment:
I believe you still need to check the existence of `mocha.options.grep` otherwise this line fails. I tested it